examples/timeSeries-train-quickdraw/sketch.js

```diff
@@ -0,0 +1,221 @@
+let state = 'collection';
+let pressedOnce = true;
+let frame_count = 0;
+let datapoints;
+let sequence = [];
+
+// Training Data lenghts
+let ink_multiplier = 3;
+let num_seq = 20;
+
+function setup() {
+  // p5 js elements
+  createCanvas(600, 400);
+  background(220);
+  UI();
+
+  // set framerate to constant rate for constant data collection
+  frameRate(60);
+
+  // set the backend to either webgl or cpu
+  ml5.setBackend('webgl');
+
+  // set the options to initialize timeSeries Neural Network
+  let options = {
+    inputs: ['x','y'],
+    outputs: ['label'],
+    task: 'classification',
+    dataModality: 'spatial',
+    debug: 'true',
+    learningRate: 0.005
+  };
+
+  model = ml5.timeSeries(options);
+}
+
+function draw() {
+  // record data when the mouse is pressed inside the canvas
+  if (mouseIsPressed && pressedOnce && mouseY<400 && mouseX<600){
+    
+    // draw lines through coordinates
+    line(pmouseX, pmouseY, mouseX,mouseY);
+
+    frame_count++;
+
+    let inputs = {x: mouseX,y: mouseY};
+    sequence.push(inputs);
+    
+    if (sequence.length == num_seq*ink_multiplier){
+      pressedOnce = false;
+      frame_count = 0
+      
+      // if state is collection, add whole sequence as X, and shape as Y
+      if (state == 'collection'){
+        let target = {label: curr_shape};
+        model.addData(sequence, target);
+
+        // add to the count for each
+        counts[`${curr_shape}_datacount`] += 1;
+        updateDataCountUI()
+        
+        // reset the screen
+        background(220);
+        textSize(20)
+        fill(0);
+        text("Recording: " + curr_shape, 50,50);
+      // if prediction, classify using the whole sequence
+      } else if (state == 'prediction'){
+        model.classify(sequence, gotResults)
+
+        background(220);
+      }
+
+      // reset the sequence 
+      sequence = [];
+    }
+  } 
+  inkBar();
+}
+
+function trainModel(){
+  // normalize Data first before Training
+  model.normalizeData();
+
+  // set the number of epochs for training
+  let options = {
+    epochs: 40,
+  }
+  model.train(options,whileTraining,finishedTraining);
+
+  background(220);
+  state = 'training';
+  text("Training...", 50,50);
+  rec_circle.style('background-color', '');
+  rec_square.style("background-color",'');
+  train_but.style('background-color', '#f0f0f0');
+}
+
+function whileTraining(epoch, loss) {
+  console.log(epoch);
+}
+
+function finishedTraining() {
+  background(220)
+  text("Training Finished, Draw again to predict", 50,50);
+  state = 'prediction';
+}
+
+function gotResults(results) {
+  const label = results[0].label;
+
+  fill(0);
+  text("Prediction: " + label, 50,50);
+}
+
+// code to signify drawing can be done again
+function mouseReleased(){
+  pressedOnce = true;
+}
+
+////////////// UI Elements ////////////
+
+// code to visualize how much ink left
+function inkBar(){
+  datapoints = map(frame_count,0,ink_multiplier*num_seq, 0,num_seq)
```

Comment:
Hi professor, I am almost done with the RDP algorithm, and it works well, the code I referenced originally uses vectors, I am currently continuing this idea but do you think it might cause issues with the user's side? (i.e. they need to know use vectors only to make this work)

Reply:
The vector math is merely inside the RDP function? Then I think this shouldn't be all too confusing

Reply:
Yes you're right! I just finished implementing the RDP and its a timeSeries utility called `padCoordinates(seq_to_pad, target_seq_size)`, I specifically chose this name to introduce the idea of padding and its importance in dealing with time series data. The example is in the mouse Gesture RDP folder. 

That said should I remove the ink one entirely? thanks!